fern/pages/tutorials/letta-rag.mdx

```diff
@@ -0,0 +1,475 @@
+
+In the standard RAG approach, our application takes the lead. It fetches the relevant information from our ChromaDB database and then passes this context, along with our query, to a simple Letta agent. This method is direct, transparent, and keeps all the retrieval logic in our client application.
+
+### Step 1: Set Up the Cloud Vector Database
+
+First, we need to populate our ChromaDB Cloud database with the content of the research papers. We'll use two papers for this demo: ["Attention Is All You Need"](https://arxiv.org/abs/1706.03762) and ["BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding"](https://arxiv.org/abs/1810.04805).
+
+Before we begin, let's create a Python virtual environment to keep our dependencies isolated:
+
+```shell
+python -m venv venv
+source venv/bin/activate  # On Windows, use: venv\Scripts\activate
+```
+
+Download the research papers we'll be using:
+
+```shell
+curl -o 1706.03762.pdf https://arxiv.org/pdf/1706.03762.pdf
+curl -o 1810.04805.pdf https://arxiv.org/pdf/1810.04805.pdf
+```
+
+Now, create a `requirements.txt` file with the necessary Python libraries:
+
+```
+letta-client
+chromadb
+pypdf
+python-dotenv
+```
+
+Install them using pip:
+
+```shell
+pip install -r requirements.txt
+```
+
+Now, create a `setup.py` file. This script will load the PDFs, split them into manageable chunks, and ingest them into a ChromaDB collection named `rag_collection`.
+
+```python
+import os
+import chromadb
+import pypdf
+from dotenv import load_dotenv
+
+load_dotenv()
+
+def main():
+    # Connect to ChromaDB Cloud
+    client = chromadb.CloudClient(
+        tenant=os.getenv("CHROMA_TENANT"),
+        database=os.getenv("CHROMA_DATABASE"),
+        api_key=os.getenv("CHROMA_API_KEY")
+    )
+
+    # Create or get the collection
+    collection = client.get_or_create_collection("rag_collection")
+
+    # Ingest PDFs
+    pdf_files = ["1706.03762.pdf", "1810.04805.pdf"]
+    for pdf_file in pdf_files:
+        print(f"Ingesting {pdf_file}...")
+        reader = pypdf.PdfReader(pdf_file)
+        for i, page in enumerate(reader.pages):
+            collection.add(
+                ids=[f"{pdf_file}-{i}"],
+                documents=[page.extract_text()]
+            )
+
+    print("\nIngestion complete!")
+    print(f"Total documents in collection: {collection.count()}")
+
+if __name__ == "__main__":
+    main()
+
+```
+
+Run the script from your terminal:
+
+```shell
+python setup.py
+```
+
+This script connects to your ChromaDB Cloud instance, creates a collection, and adds the text content of each page from the PDFs as a separate document. Your vector database is now ready.
+
+### Step 2: Create a "Stateless" Letta Agent
+
+For the standard RAG approach, the Letta agent doesn't need any special tools or complex instructions. Its only job is to answer a question based on the context we provide. We can create this agent programmatically using the Letta SDK.
```

Comment:
Stateless is a difficult term to use here, because very few Letta agents are truly stateless. It takes quite a lot of work to turn off all the things we add to your agents. A more accurate term here would be "simple".